src/components/modal/Modal.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import * as selectors from '@/reducers/selectors';
+import * as actions from '@/actions/creators';
+
+const mapStateToProps = (state) => {
+  let message = 'Magic is happening...';
+  let type = 'loading';
+  const shouldDisplayModal = selectors.getShouldDisplayModal(state);
+  const errorMessage = selectors.getErrorMessage(state);
+  const alertMessage = selectors.getAlertMessage(state);
+
+  if (errorMessage) {
```

Comment:
If we continue the "alert-level", we could save only one message as an object:
```
message: {
  text: 'This is an error',
  level: 'error',
}
```
Then we wouldn't need the if-statement here but only apply text and alertLevel to the component